interfaces/builtin/dpdk_control.go

```diff
@@ -0,0 +1,49 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const dpdkControlSummary = `allows using dpdk`
+
+const dpdkControlBaseDeclarationSlots = `
+  dpdk-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const dpdkControlConnectedPlugAppArmor = `
+# Description: Allow control to dpdk.
```

Comment:
Is the interface designed to allow DPDK to do its thing or for other things to talk to a snap providing DPDK? If the answer is _the former_ then this interface should be called _dpkg-support_ and be uniquely granted to dpdk.

Reply:
The plug allows the dpdk snap to do it's thing, but it also has to allow any application installed on top of the device to be able to make use of the DPDK modules.